src/ast/nodes/IfStatement.ts

```diff
@@ -45,33 +50,43 @@ export default class IfStatement extends StatementBase {
 	}
 
 	render(code: MagicString, options: RenderOptions) {
-		const testValue = this.test.getValue();
+		// Note that unknown test values are always included
+		const testValue = this.hasUnknownTestValue ? UNKNOWN_VALUE : this.test.getValue();
 		if (
-			!this.context.treeshake ||
-			this.test.included ||
-			(testValue ? this.alternate !== null && this.alternate.included : this.consequent.included)
+			!this.test.included &&
+			(testValue ? this.alternate === null || !this.alternate.included : !this.consequent.included)
 		) {
-			super.render(code, options);
+			const singleRetainedBranch = testValue ? this.consequent : this.alternate;
+			code.remove(this.start, singleRetainedBranch.start);
+			code.remove(singleRetainedBranch.end, this.end);
+			singleRetainedBranch.render(code, options);
 		} else {
-			// if test is not included, it is impossible that alternate===null even though it is the retained branch
-			const branchToRetain = testValue ? this.consequent : this.alternate;
-			code.remove(this.start, branchToRetain.start);
-			code.remove(branchToRetain.end, this.end);
-			branchToRetain.render(code, options);
+			if (this.test.included) {
+				this.test.render(code, options);
+			} else {
+				code.overwrite(this.test.start, this.test.end, JSON.stringify(testValue));
+			}
+			if (this.consequent.included) {
+				this.consequent.render(code, options);
+			} else {
+				code.overwrite(this.consequent.start, this.consequent.end, ';');
```

Comment:
Not expecting to get this in on this PR, but how about optimizing this a little further?

```js
if (this.alternate === null || !this.alternate.included) {
  code.overwrite(this.start, this.start + 3, '');
  code.overwrite(this.consequent.start - 1, this.consequent.end, ';');
} else {
  code.insertLeft(this.test.start, '!');
  code.overwrite(this.consequent.start, this.consequent.end + 'false'.length, '');
}
```

Reply:
So if I see this correctly, if you have both the test and only one branch included, you want to render them as just two separate statements?
That's actually a good idea but I would refrain from this for this bug fix as I see some potential for introducing new bugs here. Maybe just make it a separate PR?

Reply:
It was actually meant in the cases of the test not being included to have:

```js
if (true) effect();
```

become `effect();`, and 

```js
if (true) ;
else
  effect();
```

become `effect()`.

Reply:
I can certainly look at a separate PR though!

Reply:
Also, `if (effect());` to become `effect();`.

Reply:
Ahh I see - the two statement variation would also be worthwhile here yes.

Reply:
If the test is not included, it already works this way! This is the first branch with the `code.remove` statements.

Reply:
So missing (for a future PR?) are:
* Only the test is included -> render it as a statment
* The test and one other branch that is always executed is included -> render as two statements